autoPyTorch/utils/parallel_model_runner.py

```diff
@@ -0,0 +1,203 @@
+
+            while workers_to_wait >= 1:
+                workers_to_wait -= 1
+                # We launch dask jobs only when there are resources available.
+                # This allow us to control time allocation properly, and early terminate
+                # the traditional machine learning pipeline
+                cls, future = dask_futures.pop(0)
+                status, cost, runtime, additional_info = future.result()
+
+                if status == StatusType.SUCCESS:
+                    logger.info(
+                        "Fitting {} took {} [sec] and got performance: {}.\n"
+                        "additional info:\n{}".format(cls, runtime, cost, dict_repr(additional_info))
+                    )
+                    origin: str = additional_info['configuration_origin']
+                    current_config: Union[str, dict] = additional_info['configuration']
+                    current_budget: float = additional_info['budget']
+
+                    # indicates the finished model is part of autopytorch search space
+                    if isinstance(current_config, dict):
+                        configuration = Configuration(current_search_space, current_config)  # type: ignore[misc]
+                    else:
+                        # we assume that it is a traditional model and `pipeline_configuration`
+                        # specifies the configuration.
+                        configuration = additional_info.pop('pipeline_configuration')
+
+                    run_history.add(config=configuration, cost=cost,
+                                    time=runtime, status=status, seed=seed,
+                                    starttime=starttime, endtime=starttime + runtime,
+                                    origin=origin, additional_info=additional_info)
+                    model_identifiers.append((seed, additional_info['num_run'], float(current_budget)))
+                else:
+                    if additional_info.get('exitcode') == -6:
+                        logger.error(
+                            "Traditional prediction for {} failed with run state {},\n"
+                            "because the provided memory limits were too tight.\n"
+                            "Please increase the 'ml_memory_limit' and try again.\n"
+                            "If you still get the problem, please open an issue\n"
+                            "and paste the additional info.\n"
+                            "Additional info:\n{}".format(cls, str(status), dict_repr(additional_info))
+                        )
+                    else:
+                        logger.error(
+                            "Traditional prediction for {} failed with run state {}.\nAdditional info:\n{}".format(
+                                cls, str(status), dict_repr(additional_info)
+                            )
+                        )
+                    model_identifiers.append(None)
```

Comment:
can we move this line right after the `else` in L176 as it was hard to see which indent level this line lies?

Reply:
sure.